delivery_carrier_location/tests/test_delivery_carrier_location.py

```diff
@@ -0,0 +1,149 @@
+        )
+        cls.paris_06 = cls.env["res.city.zip"].create(
+            {"name": "75006", "city_id": cls.paris.id}
+        )
+        cls.paris_07 = cls.env["res.city.zip"].create(
+            {"name": "75007", "city_id": cls.paris.id}
+        )
+        cls.paris_08 = cls.env["res.city.zip"].create(
+            {"name": "75008", "city_id": cls.paris.id}
+        )
+        cls.paris_09 = cls.env["res.city.zip"].create(
+            {"name": "75009", "city_id": cls.paris.id}
+        )
+        cls.paris_10 = cls.env["res.city.zip"].create(
+            {"name": "75010", "city_id": cls.paris.id}
+        )
+        cls.paris_11 = cls.env["res.city.zip"].create(
+            {"name": "75011", "city_id": cls.paris.id}
+        )
+        # Disable all other delivery methods
+        cls.env["delivery.carrier"].search([]).write({"active": False})
+        # Create delivery methods
+        cls.product = cls.env["product.product"].create({"name": "Delivery"})
+        cls.carrier_paris_01_05 = cls.env["delivery.carrier"].create(
+            {
+                "name": "Delivery in Paris 01-05",
+                "product_id": cls.product.id,
+                "delivery_type": "fixed",
+                "fixed_price": 20,
+                "sequence": 10,
+                "country_ids": [(4, cls.france.id)],
+                "zip_ids": [
+                    (4, cls.paris_01.id),
+                    (4, cls.paris_02.id),
+                    (4, cls.paris_03.id),
+                    (4, cls.paris_04.id),
+                    (4, cls.paris_05.id),
+                ],
+            }
+        )
+        cls.carrier_paris_06_11 = cls.env["delivery.carrier"].create(
+            {
+                "name": "Delivery in Paris 06-11",
+                "product_id": cls.product.id,
+                "delivery_type": "fixed",
+                "fixed_price": 15,
+                "sequence": 10,
+                "country_ids": [(4, cls.france.id)],
+                "zip_ids": [
+                    (4, cls.paris_06.id),
+                    (4, cls.paris_07.id),
+                    (4, cls.paris_08.id),
+                    (4, cls.paris_09.id),
+                    (4, cls.paris_10.id),
+                    (4, cls.paris_11.id),
+                ],
+            }
+        )
+        cls.carrier_france = cls.env["delivery.carrier"].create(
+            {
+                "name": "Delivery in France",
+                "product_id": cls.product.id,
+                "delivery_type": "fixed",
+                "fixed_price": 25,
+                "sequence": 20,
+                "country_ids": [(4, cls.france.id)],
+            }
+        )
+
+    def _get_available_carriers(self, partner):
+        return self.env["delivery.carrier"].search([]).available_carriers(partner)
+
+    def test_00_partner_zip_id(self):
+        # Partner living in paris 03, with zip_id set
+        partner = self.env["res.partner"].create(
```

Comment:
can't you use existing demo partners and save some tree? :)

Reply:
I'd still need to write on them to set zip_id (not in core) so not sure it'll save anything :sweat_smile: